src/Microsoft.AspNet.Mvc.Core/Rendering/Html/DefaultHtmlGenerator.cs

```diff
@@ -123,7 +123,7 @@ public virtual TagBuilder GenerateAntiForgery([NotNull] ViewContext viewContext)
                 metadata,
                 name,
                 value: "true",
-                useViewData: !explicitValue,
+                useViewData: (metadata == null && !explicitValue),
```

Comment:
is that needed w/ following comment about 20 lines up in the `metadata != null` block?
``` c#
// CheckBoxFor() case. That API does not support passing isChecked directly.
```